lib/internal/modules/esm/get_format.js

```diff
@@ -1,21 +1,27 @@
 'use strict';
+
 const {
   RegExpPrototypeExec,
   ObjectPrototypeHasOwnProperty,
   PromisePrototypeThen,
   PromiseResolve,
+  StringPrototypeIndexOf,
   StringPrototypeCharCodeAt,
   StringPrototypeSlice,
 } = primordials;
 const { basename, relative } = require('path');
 const { getOptionValue } = require('internal/options');
 const {
   extensionFormatMap,
+  getFormatOfExtensionlessFile,
   mimeToFormat,
 } = require('internal/modules/esm/formats');
 
 const experimentalNetworkImports =
   getOptionValue('--experimental-network-imports');
+const typeFlag = getOptionValue('--experimental-type');
+// The next line is where we flip the default to ES modules someday.
+const defaultType = typeFlag === 'module' ? 'module' : 'commonjs';
```

Comment:
This seems to be the better naming IMO. Using the term "default" to describe this mode switch I think may be much more clarifying since this doesn't override existing configurations.

Reply:
https://github.com/nodejs/node/issues/49541